pkg/query-service/constants/constants.go

```diff
@@ -74,6 +75,17 @@ var DEFAULT_FEATURE_SET = model.FeatureSet{
 	TimestampSort: IsTimestampSortFeatureEnabled(),
 }
 
+func GetContextTimeout() time.Duration {
+	contextTimeoutStr := os.Getenv("CONTEXT_TIMEOUT")
+	contextTimeoutInt, err := strconv.ParseInt(contextTimeoutStr, 10, 0)
+	if err != nil {
+		return 60 // seconds
+	}
+	return time.Duration(contextTimeoutInt)
```

Comment:
This is incorrect. These are nanoseconds https://go.dev/play/p/1iPLc1PEV1_6. Use [`time.ParseDuration`](https://pkg.go.dev/time#ParseDuration) Instead which supports the values such as "120s", "3m", "1h", etc...

Reply:
Thanks for the comment, I fixed it for seconds. Do we want to support more than just seconds?

Reply:
We are already multiplying by second [here](https://github.com/SigNoz/signoz/blob/dd25ad95c77df8089aeaa9025dafc54b30eea3a0/ee/query-service/app/server.go#L405). 
I have tested this in my [PR](https://github.com/SigNoz/signoz/pull/2586), the pervious changes of this PR were fine.

@daniel-t4e Did you test your changes by running?

Reply:
@makeavish I did, i also runned all the tests

Reply:
Both the PRs mentioned the return type as `time.Duration` and commented that returned values are seconds, the incorrect part in the PR. Since you changed the type from `int` to `time.Duration` my assumption was you were going to use this directly without having to convert or multiply it somewhere else. Either change the type to `int` with the original changes or if you want to use `time.Duration`, remove any conversion elsewhere and keep the detail in one place in the code.

Reply:
Ah sorry, I misunderstood. I fixed this issue. The conversion all happens in the creation of the ContextTimeout